hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/HoodieDataSourceHelpers.java

```diff
@@ -88,6 +97,16 @@ public static String latestCommit(HoodieStorage storage, String basePath) {
     return timeline.lastInstant().get().getTimestamp();
   }
 
+  public static String latestCommitCompletionTime(FileSystem fs, String basePath) {
+    HoodieTimeline timeline = allCompletedCommitsCompactions(fs, basePath);
+    return timeline.lastInstant().get().getCompletionTime();
+  }
+
+  public static String latestCommitCompletionTime(HoodieStorage storage, String basePath) {
+    HoodieTimeline timeline = allCompletedCommitsCompactions(storage, basePath);
+    return timeline.lastInstant().get().getCompletionTime();
+  }
```

Comment:
Similar here, this returns the completion time of the commit which has the largest instant time, which might not be the max completion time on the timeline.

Reply:
Should these be moved to test utils if they are only used by the tests?

Reply:
Yes, I think it makes sense. I'll fix the completion time logic now, and will file a jira to move them to test utils because this will require changing a lot of classes.

Reply:
I moved these to `org.apache.hudi.testutils.DataSourceTestUtils`